linode/framework_provider_config.go

```diff
@@ -290,16 +290,17 @@ func (fp *FrameworkProvider) InitProvider(
 		client.SetRetryMaxWaitTime(time.Duration(maxRetryDelayMilliseconds) * time.Millisecond)
 	}
 
-	userAgent := fp.terraformUserAgent(tfVersion, UAPrefix)
+	userAgent := fp.fwProviderUserAgent(tfVersion, UAPrefix)
 	client.SetUserAgent(userAgent)
+	fp.Meta.ProviderUserAgent = userAgent
 
 	helper.ApplyAllRetryConditions(&client)
 
-	meta.Config = lpm
-	meta.Client = &client
+	fp.Meta.Config = lpm
+	fp.Meta.Client = client
 }
 
-func (fp *FrameworkProvider) terraformUserAgent(
+func (fp *FrameworkProvider) fwProviderUserAgent(
```

Comment:
nitpick: I think this can just become `userAgent(...)` since fwProvider is implied from the struct?